### PR TITLE
Adding sample migration from mongodb with sample source plugin.

### PR DESCRIPTION
**Pull request by joshirohit100**
cc @amitgoyal @pranitjha for review.

### CODE REVIEW

File: modules/examples/apm_mongo_migrate/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# MongoDB Migrate
+
+This is a sample module to migrate data from MongoDB to Drupal8.
+
+## Requirements
+
+* This module requires php `mongodb` extension to be enabled.
+* Follow/See `http://php.net/manual/en/set.mongodb.php` to
```
Comment (amitgoyal) [inline modules/examples/apm_mongo_migrate/README.md]: Make it hyperlink and add the pecl setup command here as well.

File: modules/examples/apm_mongo_migrate/src/Plugin/migrate/source/ApmMongo.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $migration);
+    // @see settings.php for connection info.
+    $this->mongoDBConnection = Settings::get('mongodb');
```
Comment (amitgoyal) [inline modules/examples/apm_mongo_migrate/src/Plugin/migrate/source/ApmMongo.php]: May be add basic checks.

File: modules/examples/apm_mongo_migrate/README.md (hunk 1 above)
Comment (joshirohit100) [reply modules/examples/apm_mongo_migrate/README.md]: PR updated! Thanks!

Unable to make the link to open in new tab.

File: modules/examples/apm_mongo_migrate/src/Plugin/migrate/source/ApmMongo.php (hunk 2 above)
Comment (joshirohit100) [reply modules/examples/apm_mongo_migrate/src/Plugin/migrate/source/ApmMongo.php]: PR updated!

**Pull request merged**